lib/fog/hp/storage.rb

```diff
@@ -303,55 +303,37 @@ class Real
-            options[:hp_access_key] = options.delete(:hp_account_id)
+            @options[:hp_access_key] = @options.delete(:hp_account_id)
           end
-          @hp_access_key = options[:hp_access_key]
+          @hp_access_key = @options[:hp_access_key]
           unless @hp_access_key
             raise ArgumentError.new("Missing required arguments: hp_access_key. :hp_account_id is deprecated, please use :hp_access_key instead.")
           end
-          if options[:os_account_meta_temp_url_key]
+          if @options[:os_account_meta_temp_url_key]
             Fog::Logger.deprecation(":os_account_meta_temp_url_key is deprecated, and will be removed in a future release. please use the :openstack provider instead.")
-            @os_account_meta_temp_url_key = options.delete(:os_account_meta_temp_url_key)
+            @os_account_meta_temp_url_key = @options.delete(:os_account_meta_temp_url_key)
           end
-          @hp_secret_key = options[:hp_secret_key]
-          @hp_auth_uri   = options[:hp_auth_uri]
-          @hp_cdn_ssl    = options[:hp_cdn_ssl]
-          @connection_options = options[:connection_options] || {}
-          ### Set an option to use the style of authentication desired; :v1 or :v2 (default)
-          ### A symbol is required, we should ensure that the value is loaded as a symbol
-          auth_version = options[:hp_auth_version] || :v2
-          auth_version = auth_version.to_s.downcase.to_sym
+          @hp_secret_key = @options[:hp_secret_key]
+          @hp_auth_uri   = @options[:hp_auth_uri]
+          @hp_cdn_ssl    = @options[:hp_cdn_ssl]
+          @connection_options = @options[:connection_options] || {}
 
           ### Pass the service name for object storage to the authentication call
-          options[:hp_service_type] ||= " object-store"
-          @hp_tenant_id = options[:hp_tenant_id]
-          @hp_avl_zone  = options[:hp_avl_zone]
+          @options[:hp_service_type] ||= " object-store"
+          @hp_tenant_id = @options[:hp_tenant_id]
+          @hp_avl_zone  = @options[:hp_avl_zone]
 
-          ### Make the authentication call
-          if (auth_version == :v2)
-            # Call the control services authentication
-            credentials = Fog::HP.authenticate_v2(options, @connection_options)
-            # the CS service catalog returns the cdn endpoint
-            @hp_storage_uri = credentials[:endpoint_url]
-            @hp_cdn_uri  = credentials[:cdn_endpoint_url]
-            @credentials = credentials
-          else
-            # Call the legacy v1.0/v1.1 authentication
-            credentials = Fog::HP.authenticate_v1(options, @connection_options)
-            # the user sends in the cdn endpoint
-            @hp_storage_uri = options[:hp_auth_uri]
-            @hp_cdn_uri  = options[:hp_cdn_uri]
-          end
-
-          @auth_token = credentials[:auth_token]
+          authenticate(@options)
```

Comment:
Why do you need to pass `@options` here?
I think you can just use `@options` in the context of `authenticate`, right?